### DISCUSSION TITLE
Pocketbase Auth Sessions

### DISCUSSION THREAD

**Discussion by jnatmorris** [pocketbase/pocketbase#1670] [Q&A] [answered]
Hey yall just a quick question, does pocketbase support sessions in the JS SDK or have guidance for REST API on this topic? Such that a refresh does not log a user out after having already logged in with email and password.

I can't seem to find anything in the documentation nor in other discussions about it, so any resources are much appreciated. 

**Edit:** I've seen hints about it here and there, but have not seen a definite answers to this question.  

Cheers!

**Comment by jnatmorris** [comment]
Update, after much google-ing I've come across the methods `exportToCookie` and `loadFromCookie`. Still trying to get it to work if someone comes along and knows. 

I've gotten to where `exportToCookie` works, and where local storage under the key pocketbase_auth is updated, but having trouble getting the data back out. Oddly enough, probably due to me, the values disappear or page reload when `loadFromCookie` runs but not understanding where this data goes. Thought it would go into the authstore, but not seeing it.  

Edit: Found part of the answer [here](https://github.com/pocketbase/pocketbase/discussions/728#discussioncomment-3807036). I was under the impression that localstorage was not a suitable place to hold the token, but according to this I guess it is. Hopefully this helps others who have the same problem. Would love to still get a confirmation on all this if anyone else has insights.

**Comment by ganigeorgiev** [comment, marked as answer]
@jnatmorris PocketBase API is completely stateless and doesn't have sessions in the traditional sense.

You can simulate "session"-like behavior (eg. if you need to integrate with middleware/3rd party server app) by using cookies. SSR examples you can find in https://github.com/pocketbase/js-sdk#ssr-integration. **You don't need auth cookie/"sessions" if your app is entirely client-side (eg. spa) and interacts directly with PocketBase.**

By default the JS SDK stores the auth token in the LocalStorage. If you don't like this, you can provide your own custom [AuthStore](https://github.com/pocketbase/js-sdk#authstore).

If you are concerned about security, please check https://github.com/pocketbase/js-sdk#security.

**Reply by jnatmorris** [reply]
Thank you for your speedy responce. A few rapid fire questions:

1. In a SPA, is there a specified/preferred way (pb method) to get the auth token and model from Localstorage, or is that something the dev should take care of?
2. If one is wants to use middleware, should they go with the SSR approach too?

**Reply by ganigeorgiev** [reply]
1. You can access the token and model from `pb.authStore.token` and `pb.authStore.model`. You can find more information in the JS SDK readme - https://github.com/pocketbase/js-sdk#authstore.

2. It is hard to say since it depends how you are planning to interact with the PocketBase APIs. Usually you have to use the cookie approach if the middleware/3rd party app is the one that make requests to the PocketBase server (eg. you have a request flow like: "user <-> node server <-> PocketBase").

**Reply by jnatmorris** [reply]
Thanks for the info, was able to get it all working. 

For anyone coming to this through google, my working solution ended up being, as per my project specifications, re-creating a bunch of the functionality by using the REST API and writing and reading from localstorage myself. The API docs I used were within the admin dashboard under API Preview when looking at the users collection. But if one can use the JS SDK, I'd say that's a simpler option.
